src/App.tsx

```diff
@@ -1,11 +1,23 @@
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = useCallback((movie: Movie) => {
+    setMovies(prevMovies => {
+      const isOnTheList = prevMovies.some(m => m.imdbId === movie.imdbId);
```

Comment:
try using better naming
```suggestion
      const isOnTheList = prevMovies.some(existingMovie => existingMovie.imdbId === movie.imdbId);
```